### DISCUSSION TITLE
Why is cypress opening up a new tab unexpectedly?

### DISCUSSION THREAD

**Discussion by sinthujan24** [cypress-io/cypress#14758] [Questions and Help]
Manually, when a video tile is clicked, it opens on the same window but on cypress it's opening on a new tab. 

Clicking on the video opens a new tab (Expected to open the video player on the same screen)
```
const urls = [`https://cat-press.com/cat-movies/firework`]
  urls.forEach((url) => {
    it(`Should load the videos and autoplay one: ${url}`, () => {
      cy.visit(url)
      // Scroll to feed
      cy.get('.cy-grid').scrollIntoView()
      // Should hover over to the video at index 1
      cy.get(`.cy-thumbnail-link`).eq(1).trigger('click')
     
    })
  })
```
![image](https://user-images.githubusercontent.com/76002992/105928936-84e2ca80-6014-11eb-9331-0813bc53fc22.png)

**Comment by portyu9** [comment]
Hi @sinthujan24 

This looks less like Cypress intentionally opening a new tab and more like the application's normal click handling not being initialized correctly under Cypress.

This was reproduced in Cypress issue #14757. A Cypress maintainer observed that the Firework video card had a click handler in a normal browser, but that handler was missing in the Cypress reproduction. That could explain why the link falls back to its normal navigation behavior and opens another tab.

First, use Cypress's actual action command rather than manually triggering the event:

```js
cy.get('.cy-thumbnail-link')
  .eq(1)
  .should('be.visible')
  .click()
```

`.trigger('click')` and `.click()` are not equivalent. However, `.click()` alone did not fix the original reproduction.

If the element is an `<a target="_blank">`, this can force same-tab navigation:

```js
cy.get('.cy-thumbnail-link')
  .eq(1)
  .invoke('removeAttr', 'target')
  .click()
```

but that is a workaround because it changes the application's behavior during the test.

The real issue to investigate is why the Firework click handler is not being initialized/executed normally when the page runs under Cypress.

**Reply by portyu9** [reply]
@sinthujan24 Please mark this answer as accepted if it helped, thank you.
